### PR TITLE
Support for changing the base domain

### PR DESCRIPTION
**Pull request by hardillb**
part of flowforge/flowforge#2240

## Description

Allow the domain bound to instances to be changed.

This does require the forge app to run on both old/new domain as existing instances will need access old domain

## Related Issue(s)

flowforge/flowforge#2240

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/flowforge/flowforge/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `flowforge/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `flowforge/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

File: kubernetes.js [hunk 1]
```diff
@@ -329,12 +335,37 @@ const createService = async (project, options) => {
 const createIngress = async (project, options) => {
     const prefix = project.safeName.match(/^[0-9]/) ? 'srv-' : ''
 
+    const url = new URL(project.url)
```
Comment (knolleary) [inline kubernetes.js:338]: On line 244 you guard against `project.url` being undefined. Do we need a similar check here?

**Comment by knolleary** [comment]
Can you explain how `alternateHostnames` is set and how that'll get used?

File: kubernetes.js (hunk 1 above)
Comment (hardillb) [reply kubernetes.js:338]: No, because it is set and saved on line 320-321 so should be in place when the `createIngress()` is called on 375-377

**Comment by hardillb** [comment]
> Can you explain how alternateHostnames is set and how that'll get used?

They are not set anywhere yet, that is planning for when we allow custom domain names. 

Looks like I forgot to remove it. Will clean up.

**Review by knolleary — APPROVED** [review]

File: kubernetes.js (hunk 1 above)
Comment (knolleary) [reply kubernetes.js:338]: Thanks - missed that bit.

**Pull request merged**